sonic_fan/fan_base.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+#
+# fan_base.py
+#
+# Abstract base class for implementing platform-specific
+#  FAN control functionality for SONiC
+#
+
+try:
+    import abc
+except ImportError as e:
+    raise ImportError (str(e) + " - required module not found")
+
+class FanBase(object):
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
```

Comment:
can you remove the "abstractmethod" restriction? understood that you are following the convention in other plugin base classes. But with this restriction, it requests all the vendors to implement these functions or at least write some stub functions if they don't want to implement them, otherwise, it will fail when loading the plugins on the vendor's platform. We are in the progress to replace plugins with new platform API, so it's highly possible that some vendors don't want to implement these new plugins, so please consider this.

And I suggest you think about using new platform API instead of adding new plugins,  eventually we will use new platform API, put efforts on adding new plugins is kind of waste in my view.

Reply:
Agree with @keboliu -- plugins have been deprecated in favor of the new object-oriented platform 2.0 API. Developing new plugins seems unnecessary as they should no longer be used by platform developers.